### ISSUE TITLE
feat(rules): enforce document-level uniqueness for meta charset / description / http-equiv coexistence

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3634]
## Summary

Document-level uniqueness for `<meta>` elements:

- `<meta charset>` must not appear more than once
- `<meta name="description">` must not appear more than once
- `<meta charset>` and `<meta http-equiv="content-type">` must not coexist

## Resolution

The three constraints were implemented in PR #3677 as virtual rules in
`markuplint:html-standard` (`no-duplicate-charset`,
`no-duplicate-description`, `no-charset-http-equiv-coexist`), each
backed by `disallowed-element` with sibling-selector patterns.

The benchmark continued to report the affected fixtures as `nu-only`
because `tests/external/bench/config.ts` enables a curated rule list
and does not `extends` the preset, so virtual rules added under
`nodeRules[]` were invisible to the bench. Mirroring the same
`disallowed-element` selectors into `bench/config.ts` flips the
fixtures to `match-error`. See the gotcha note in
[`tests/external/CLAUDE.md`](https://github.com/markuplint/markuplint/blob/dev/tests/external/CLAUDE.md#adding-a-preset-virtual-rule-gotcha)
for the workflow.

Visible `<main>` uniqueness was originally listed here too, but
`no-duplicate-visible-main` already covered it (both fixtures
`match-error`); it was dropped from scope before this fix.

## Spec reference

- https://html.spec.whatwg.org/multipage/semantics.html#the-meta-element

## Impact

See the cross-reference block below for the fixture list and the
post-fix verdict tally.

## Benchmark cross-reference

_Sync'd from `tests/external/snapshots/` (submodule=1429313, nu-validator@sha256:59b0e97e2664, markuplint=5.0.0-rc.4, generated=2026-04-27 06:10:57 UTC)._

Filter: `meta.*multiple-charset|meta.*duplicate-charset|meta.*multiple-description|charset-and-(content-type|http-equiv)|multiple-visible-main|multiple-main-visible` → **7** fixtures. Verdict tally: match-error=7, match-clean=0, ml-only=0, nu-only=0, nu-over=0.

All meta-uniqueness fixtures are `match-error` after PR #3677 introduced the preset virtual rules and the bench config mirrored their selectors. `<main>` uniqueness is covered separately by `no-duplicate-visible-main`.

| fixture | verdict | nu id / ml rule |
| --- | --- | --- |
| `html/assertions/multiple-main-visible-novalid.html` | match-error | — |
| `html/elements/meta/charset-and-content-type-novalid.html` | match-error | — |
| `html/elements/meta/charset-and-http-equiv-novalid.html` | match-error | — |
| `html/elements/meta/duplicate-charset-novalid.html` | match-error | — |
| `html/elements/meta/multiple-charset-novalid.html` | match-error | — |
| `html/elements/meta/multiple-description-novalid.html` | match-error | — |
| `html/elements/multiple-visible-main-novalid.html` | match-error | — |

Reproduce: see [`tests/external/CLAUDE.md`](https://github.com/markuplint/markuplint/blob/dev/tests/external/CLAUDE.md) "Auditing a claim against the benchmark".

**Comment by YusukeHirao** [comment]
v5.0 推奨（C群）として #3684 に整理。missed error ~9件。v5.x マイナーでも追加可。

**Comment by YusukeHirao** [comment]
## nu-html-checker 実出力での検証

Docker 経由で nu-html-checker を実行して確認:

### テスト HTML
```html
<meta charset="utf-8">
<meta charset="utf-8">
<meta http-equiv="content-type" content="text/html; charset=utf-8">
<meta name="description" content="x">
<meta name="description" content="y">
<main>a</main>
<main>b</main>
```

### nu-html-checker 出力
6 件の error:
- encoding mismatch
- charset must be utf-8
- duplicate meta charset
- charset + http-equiv coexistence 禁止
- duplicate meta description
- duplicate visible main

### markuplint 出力
2 件の error のみ:
- `invalid-attr` で charset value
- `no-duplicate-visible-main` で second main

**4/6 未検出**: duplicate meta charset / charset+http-equiv coexistence / duplicate meta description

### 判定: CONFIRMED

ドキュメントレベル一意性制約（meta 系）は未実装。新規ルール追加が必要。

**Comment by YusukeHirao** [comment]
## ⚠️ 過去コメントは未検証

本 Issue の 2026-04-14 以前のコメントは nu-html-checker を実行せずに書かれた推測ベースです。以下が Docker 実測結果です。

## 実測結果（2026-04-14）

### 4 件の独立検証

| # | 制約 | nu-html-checker | markuplint | Gap |
|---|-----|----------------|-----------|-----|
| 1 | `<meta charset>` 重複 | `A document must not include more than one "meta" element with a "charset" attribute.` | （検出なし） | **YES** |
| 2 | `<meta charset>` + `<meta http-equiv="content-type">` 共存 | `A document must not include both a "meta" element with an "http-equiv" attribute whose value is "content-type", and a "meta" element with a "charset" attribute.` | （検出なし） | **YES** |
| 3 | `<meta name="description">` 重複 | `A document must not include more than one "meta" element with its "name" attribute set to the value "description".` | （検出なし） | **YES** |
| 4 | `<main>` 重複 | `A document must not include more than one visible "main" element.` | 「There must not be more than one visible "main" element in a document」(`no-duplicate-visible-main`) | NO（既に検出済み） |

## 真のスコープ

3 件の未実装:
- `<meta charset>` 重複
- `<meta charset>` と `<meta http-equiv="content-type">` の共存
- `<meta name="description">` 重複（他の name 値も要確認: author, generator, keywords, theme-color, viewport 等）

## 実装方針

`@markuplint/rules` に新ルール `no-duplicate-meta` または既存 `no-duplicate-*` 系の拡張。
